azurerm/internal/services/machinelearning/machine_learning_inference_cluster_resource.go

```diff
@@ -0,0 +1,495 @@
+	if err != nil {
+		return fmt.Errorf("error creating Inference Cluster %q in workspace %q (Resource Group %q): %+v", name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+	if err := future.WaitForCompletionRef(ctx, mlComputeClient.Client); err != nil {
+		return fmt.Errorf("error waiting for creation of Inference Cluster %q in workspace %q (Resource Group %q): %+v", name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+	resp, err := mlComputeClient.Get(ctx, workspaceID.ResourceGroup, workspaceID.Name, name)
+	if err != nil {
+		return fmt.Errorf("error retrieving Inference Cluster Compute %q in workspace %q (Resource Group %q): %+v", name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("cannot read Inference Cluster ID %q in workspace %q (Resource Group %q) ID", name, workspaceID.Name, workspaceID.ResourceGroup)
+	}
+
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	id := parse.NewInferenceClusterID(subscriptionId, workspaceID.ResourceGroup, workspaceID.Name, name)
+	d.SetId(id.ID())
+
+	return resourceAksInferenceClusterRead(d, meta)
+}
+
+func resourceAksInferenceClusterRead(d *schema.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	aksClient := meta.(*clients.Client).Containers.KubernetesClustersClient
+	poolsClient := meta.(*clients.Client).Containers.AgentPoolsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.InferenceClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", id.ComputeName)
+
+	// Check that Inference Cluster Response can be read
+	resp, err := mlComputeClient.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.ComputeName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error making Read request on Inference Cluster %q in Workspace %q (Resource Group %q): %+v",
+			id.ComputeName, id.WorkspaceName, id.ResourceGroup, err)
+	}
+
+	// Retrieve Machine Learning Workspace ID
+	mlResp, err := mlWorkspacesClient.Get(ctx, id.ResourceGroup, id.WorkspaceName)
+	if err != nil {
+		return err
+	}
+	d.Set("machine_learning_workspace_id", mlResp.ID)
+
+	// Retrieve AKS Cluster ID
+	aksResp, err := aksClient.ListByResourceGroup(ctx, id.ResourceGroup)
+	if err != nil {
+		return err
+	}
+
+	unparsedAksId := *(aksResp.Values()[0].ID)
+
+	// Retrieve AKS Cluster name and Node pool name from ID
+	aksId, err := parse.KubernetesClusterID(unparsedAksId)
+	if err != nil {
+		return err
+	}
+
+	nodePoolList, _ := poolsClient.List(ctx, id.ResourceGroup, aksId.ManagedClusterName)
+	unparsedPoolId := *(nodePoolList.Values()[0].ID)
+	poolId, err := parse.NodePoolID(unparsedPoolId)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we should remove the node pool (and the configuration based on the node pool) from the configuration. This is not used when attaching a cluster from the [commandline](https://docs.microsoft.com/en-us/cli/azure/ml/computetarget/attach?view=azure-cli-latest), so I expect the configuration not necessary to the functioning of the inference cluster. WDYT?

For a reference I use both the `attach` and [`update` docs](https://docs.microsoft.com/en-us/cli/azure/ml/computetarget/update?view=azure-cli-latest#az_ml_computetarget_update_aks) for an AKS compute target, so maybe we should expose some other knobs to effectively update the configuration.

Reply:
Hmm... Yeah I am not sure how to determine the subnet id from the existing AKS service in another way. The other thing for what I used it is to determine agent count and vm size, which I thought you could infer from the Azure Kubernetes node pool, but maybe you should do that differently. You might even need to use them as input parameters (if not inferred from a node pool), which would leave us with the problem of how to determine the subnet id from an existing AKS.